xla/backends/gpu/runtime/command_buffer_thunk.cc

```diff
@@ -238,6 +245,14 @@ absl::Status CommandBufferThunk::ExecuteOnStream(const ExecuteParams& params) {
 
   absl::MutexLock lock(cmd_buffer->mutex);
 
+  // warm up iteration, run through thunks if they are present.
+  if (!cmd_buffer->warmup_done && thunks_) {
+    VLOG(2) << "Executing warm up iteration of command buffer thunk";
+    TF_RETURN_IF_ERROR(thunks_->ExecuteOnStream(params));
```

Comment:
I'm a bit concerned that we holding the command buffer mutex while calling ExecuteOnStream. I'd release the mutex before executing the thunks, then re-acquire to set the flag. @ezhulenev What you think?

Reply:
Eugene Zhulenev, Yesterday 3:28 PM
I think that's ok, just for the warmup